sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultRestoreOperation.cs

```diff
@@ -15,6 +15,9 @@ namespace Azure.Security.KeyVault.Administration
     public class KeyVaultRestoreOperation : Operation<KeyVaultRestoreResult>
     {
         internal readonly RestoreOperationInternal<AzureSecurityKeyVaultAdministrationFullRestoreOperationHeaders, KeyVaultRestoreResult, RestoreDetailsInternal> _operationInternal;
+        internal readonly RestoreOperationInternal<AzureSecurityKeyVaultAdministrationPreFullRestoreOperationHeaders, KeyVaultRestoreResult, RestoreDetailsInternal> _preOperationInternal;
```

Comment:
I don't think this will scale well. I appreciate they are basically the same code, but you could just as easily copy and paste the code as a new class. This allows any changes on either endpoint to be reflected in the types here separately. If you return the same class from the client method, we're locked into supported any distinct changes to either endpoint that may be confusing or even conflict.

Blocking on separation of these two LROs. @vcolin7, if other languages did this, I don't think they should. The SDK architects tend to avoid smooshing too much together for this very reason. It's also why, for example, the Azure REST API board "strongly disproves" of REST APIs that return an array instead of an object with a sole array: you can only expand the latter - not the former.

Reply:
I'll take a look at what we did in Java. It's possible I tried to re-use one of the models from the LROs.

Reply:
@vcolin7 I looked at your PR and noticed you’re reusing the same models. I encountered an issue with reusing the same model here because the `Retry-After` response header was of type `long` for the pre-operations. However, this has been changed to `int32` in [this PR](https://github.com/Azure/azure-rest-api-specs/pull/30115), allowing us to reuse them in C# as well.

On the other hand, based on what @heaths mentioned, I now see how that approach will cause headaches in the future. Therefore, I will be creating new classes for the pre-restore and pre-backup operations. For now, I want to clarify that both types will essentially be identical, with the same property names and types.

Reply:
I think any strongly-typed language should. It's safer and doesn't paint us into a corner.